include/vsg/maths/quat.h

```diff
@@ -132,17 +143,21 @@ namespace vsg
 
             double angle_radians = acos(dot_pd / div);
 
-            value_type inversenorm = 1.0 / len;
-            value_type coshalfangle = cos(0.5 * angle_radians);
-            value_type sinhalfangle = sin(0.5 * angle_radians);
+            value_type inversenorm = one / len;
+            value_type coshalfangle = cos(half * angle_radians);
+            value_type sinhalfangle = sin(half * angle_radians);
 
             x = axis.x * sinhalfangle * inversenorm;
             y = axis.y * sinhalfangle * inversenorm;
             z = axis.z * sinhalfangle * inversenorm;
             w = coshalfangle;
         }
 
-        explicit operator bool() const noexcept { return value[0] != 0.0 || value[1] != 0.0 || value[2] != 0.0 || value[3] != 0.0; }
+        explicit operator bool() const noexcept
+        {
+            static constexpr value_type zero = static_cast<value_type>(1.0);
```

Comment:
Perhaps this might be best written as:

static constexpr value_type zero = static_cast<value_type>(0.0);

:-)